.github/workflows/test.yml

```diff
@@ -2,27 +2,16 @@ on: [push, pull_request]
 name: Test
 jobs:
   test:
-    strategy:
-      matrix:
-        go-version: [1.14.x, 1.15.x, 1.16.x, 1.17.x]
-        os: [ubuntu-latest, macos-latest, windows-latest]
-    runs-on: ${{ matrix.os }}
+    name: Test
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
     steps:
-    - name: Install Go
-      uses: actions/setup-go@v2
+    - name: Checkout
+      uses: actions/checkout@v4
+    - name: Setup Golang
+      uses: actions/setup-go@v5
       with:
-        go-version: ${{ matrix.go-version }}
-    - name: Cache-Go
```

Comment:
caching is now handled natively within setup-go

Reply:
Noted. I will have to port that to my other projects.